compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -136,12 +136,17 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         for ambiguity_error in &self.ambiguity_errors {
             let diag = self.ambiguity_diagnostics(ambiguity_error);
             if ambiguity_error.warning {
-                let NameBindingKind::Import { import, .. } = ambiguity_error.b1.0.kind else {
-                    unreachable!()
-                };
+                let root_id =
+                    if let NameBindingKind::Import { import, .. } = ambiguity_error.b1.0.kind {
+                        import.root_id
+                    } else if ambiguity_error.kind == AmbiguityKind::ExternalCrate {
+                        krate.id
```

Comment:
```suggestion
                        CRATE_NODE_ID
```
Would be more clear.

Also `assert_eq!(ambiguity_error.kind, AmbiguityKind::ExternalCrate);` in `else` instead of `else if`.